src/components/DomainField/DomainField.js

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useMemo } from 'react'
+import { Field, GU, TextInput, useTheme } from '@aragon/ui'
+import Check from './Check'
+
+export const DOMAIN_CHECK = Symbol('DOMAIN_CHECK')
+export const DOMAIN_LOADING = Symbol('DOMAIN_LOADING')
+export const DOMAIN_ERROR = Symbol('DOMAIN_ERROR')
+export const DOMAIN_NONE = Symbol('DOMAIN_NONE')
+
+// Filter a subdomain
+function filterSubdomain(subdomain) {
+  return subdomain
+    .trim()
+    .toLowerCase()
+    .replace(/[^a-z0-9]/g, '')
+    .slice(0, 30)
+}
+
+// Get the subdomain from a complete domain
+function getSubdomain(domain = '', domainEnd = '') {
+  const lastIndex = domain.lastIndexOf(domainEnd)
+  return lastIndex + domainEnd.length === domain.length
+    ? domain.slice(0, lastIndex)
+    : domain
+}
+
+function DomainField({
+  domainEnd = '.aragonid.eth',
+  label = 'Name of the organization',
+  onChange,
+  placeholder = 'Type an organization name',
+  status = DOMAIN_CHECK,
+  value,
+  ...props
+}) {
+  const theme = useTheme()
+
+  const handleInputChange = useCallback(
+    event => {
+      onChange(filterSubdomain(event.target.value) + domainEnd)
+    },
+    [domainEnd, onChange]
+  )
+
+  const subdomain = useMemo(() => getSubdomain(value, domainEnd), [
+    value,
+    domainEnd,
+  ])
+
+  return (
+    <div
+      css={`
+        display: flex;
+        align-items: center;
+        flex-grow: 1;
+      `}
+      {...props}
+    >
+      <div
+        css={`
+          display: flex;
+          width: 100%;
+        `}
+      >
+        <Field label={label} css="width: 100%">
+          <TextInput
+            wide
+            placeholder={placeholder}
+            value={subdomain}
+            onChange={handleInputChange}
+            adornment={
+              <div
+                css={`
+                  display: flex;
+                  align-items: center;
+                  height: calc(100% - 2px);
+                  margin: 1px 0;
+                  padding: 0 ${2 * GU}px;
+                  border-left: 1px solid ${theme.border};
+                  background: ${theme.surface};
+                `}
+              >
+                {domainEnd}
```

Comment:
It would be nice to make this its own input field, and controllable by the user whether it's readonly or not.

When we use this in `OpenOrg`, it'd be nice to let users change the `.aragonid.eth` domain (e.g. in case they'd like to visit `governance.aragonproject.eth`.

Reply:
I don’t think we allow sub-subdomains in `.aragonid.eth`? If that’s the case, what would you think of removing the `.aragonid.eth` suffix as soon as a `.` appears in the entered domain? /cc @dizzypaty

Reply:
Good idea : )

Reply:
Done in https://github.com/aragon/aragon/pull/921/commits/0e45e75b19d11773a5d77d786a796ed144bc1e2f https://github.com/aragon/aragon/pull/921/commits/8a879a83efa8a156e0f21e50d0537c34dffa598e https://github.com/aragon/aragon/pull/921/commits/b91a73a6c1260e1149ce0202be8d7c2d27357387